src/GovUk.Education.ExploreEducationStatistics.Content.Services/Strategies/AnalyticsWritePublicCsvDownloadStrategy.cs

```diff
@@ -0,0 +1,64 @@
+#nullable enable
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using GovUk.Education.ExploreEducationStatistics.Analytics.Common.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Common.Services;
+using GovUk.Education.ExploreEducationStatistics.Common.Utils;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Interfaces;
+using GovUk.Education.ExploreEducationStatistics.Content.Services.Requests;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+
+namespace GovUk.Education.ExploreEducationStatistics.Content.Services.Strategies;
+
+public class AnalyticsWritePublicCsvDownloadStrategy(
+    IAnalyticsPathResolver analyticsPathResolver,
+    DateTimeProvider dateTimeProvider,
+    ILogger<AnalyticsWritePublicCsvDownloadStrategy> logger
+    ) : IAnalyticsWriteStrategy
+{
+    public Type RequestType => typeof(CaptureCsvDownloadRequest);
+
+    public async Task Report(IAnalyticsCaptureRequestBase request, CancellationToken cancellationToken)
+    {
+        if (request is not CaptureCsvDownloadRequest csvDownloadRequest)
```

Comment:
I think that seeing this pattern repeated again and again (and I'm adding more into Public API at the mo), this is a prime candidate to have the majority of this stripped out and put somewhere common.  Not something for this PR though!